test/test_broker.hpp

```diff
@@ -1585,20 +1642,36 @@ class test_broker {
                 retains_.erase(topic);
             }
             else {
+                std::shared_ptr<as::steady_timer> tim_message_expiry;
+                if(message_expiry_interval) {
+                    tim_message_expiry = std::make_shared<as::steady_timer>(ioc_, message_expiry_interval.value());
+                    tim_message_expiry->async_wait(
+                      [this, topic = topic, wp = std::weak_ptr<as::steady_timer>(tim_message_expiry)]
+                      (boost::system::error_code const& ec) {
+                      if (wp.lock()) {
```

Comment:
```suggestion
                      if (auto sp = wp.lock()) {
```

Please do as I mentioned way.
shared_ptr is temporary variable. It lose lock immediately. lvalue is needed.

Reply:
Ok. But you will warning variable is unused?

Reply:
I don't think so. See https://wandbox.org/permlink/1CpMqC1i6Yq7hBrw
Only f is warned because foo has trivial destructor. But b is not warned because it has user defined destructor. Even if the destructor is empty, it is non trivial.

std::shared_ptr has destructor. So if compiler is smart enough, no unuserd variable warning outputs.